wagtail/contrib/settings/registry.py

```diff
@@ -46,31 +47,26 @@ def is_shown(self, request):
 
 class SiteSettingAdminURLFinder(ModelAdminURLFinder):
     def construct_edit_url(self, instance):
-        return reverse(
-            "wagtailsettings:edit",
-            args=[
-                self.model._meta.app_label,
-                self.model._meta.model_name,
-                instance.site_id,
-            ],
+        return get_edit_setting_url(
+            self.model._meta.app_label,
+            self.model._meta.model_name,
+            instance.site_id,
+            locale=get_locale_for(instance=instance),
```

Comment:
Because we don't always get the locale from an instance. We can get it from a request or/and a model: https://github.com/wagtail/wagtail/pull/9390/files#diff-1b8511bbd47182a3cbc4099e5d8426e648c02b2e3b8b3f5d9d1cec6ac61d4a73R55-R58